client/src/components/molecules/CategoryMatcherAdder/CategoryMatcherAdder.tsx

```diff
@@ -52,74 +61,107 @@ export const CategoryMatcherAdder = ({
-        >
-          <CategorySelect
-            placeholder="Category"
-            loading={isListLoading}
-            options={data?.results?.map((d) => ({
-              label: <>
-                {d.name} {d.variant}
-                {
-                  <Tooltip title="Delete category">
-                    <Button type="text" icon={<DeleteOutlined />} onClick={ (e) => handleDelete(d.id, e)}></Button>
-                  </Tooltip>
-                }
-              </>,
-              value: d.id,
-            }))}
-            onAddOption={(name, variant) => mutate({ name, variant })}
-          />
-        </Form.Item>
+					</Button>
+				</Space>
+			}
+		>
+			<Form layout="vertical" requiredMark={false} form={form}>
+				<Form.Item
+					name="regexExpression"
+					label="REGEX to select records"
+					tooltip={`To match only this specific record you need to provide date in the beggining: ${record?.date} ${record?.description}`}
+					rules={[{ required: true, message: 'Please enter regex expression' }]}
+				>
+					<TextArea placeholder="Please enter regex expression" autoSize />
+				</Form.Item>
+				<Form.Item
+					name="category"
+					label="Category to assign"
+					rules={[{ required: true, message: 'Please select a category' }]}
+				>
+					<CategorySelect
+						placeholder="Category"
+						loading={isListLoading || categoryMatchersLoading}
+						options={data?.results?.map((d) => ({
+							label: <>
+								{d.name} {d.variant}
+								{
+									<Tooltip title="Delete category">
+										<Button type="text" icon={<DeleteOutlined />} onClick={(e) => handleDelete(d.id, e)}></Button>
+									</Tooltip>
+								}
+							</>,
+							value: d.id,
+						}))}
+						onAddOption={(name, variant) => mutate({ name, variant })}
+					/>
+				</Form.Item>
         Currently matches {regexMatchesList?.count} records:
-        <List
-          bordered
-          dataSource={regexMatchesList?.results}
-          itemLayout="horizontal"
-          renderItem={(item) => (
-            <List.Item>
-              <List.Item.Meta
-                title={item.date}
-                description={`${item.description}, ${item.amount}`}
-              />
-            </List.Item>
-          )}
-        />
-      </Form>
-    </Drawer>
-  );
+				<List
+					bordered
+					dataSource={regexMatchesList?.results}
+					itemLayout="horizontal"
+					renderItem={(item) => (
+						<List.Item>
+							<List.Item.Meta
+								title={item.date}
+								description={`${item.description}, ${item.amount}`}
+							/>
+						</List.Item>
+					)}
+				/>
+			</Form>
+			<Modal title="Remove matchers" open={removeModalOpen} okText={'Remove category'} onOk={() => {
+				handleRemoveCategory();
+				rematchCategories();
+				setRemoveModalOpen(false);
+			}}
+			onCancel={() => setRemoveModalOpen(open => !open)}>
+				<p>This category is used in the following category matchers that will also be removed:</p>
+				<Space direction="vertical">
```

Comment:
This component is starting to get a little too complex, let's extract this delete UI & logic to a separate component